### ISSUE TITLE
[Feature] Adding Tags to Simulations

### ISSUE THREAD

**Issue by cristianacarpinteiro** [inductiva/inductiva#704]
### Feature Description:
Here, I propose the introduction of a new feature to enable users to associate tags with each simulation. This feature allows for better organization, search, and analysis of simulations.

### Implementation:

I propose the following structure for adding and using tags:

```python
import inductiva 

# Create a scenario
scenario = inductiva.molecules.ProteinSolvation(pdb_id)

# Simulate the scenario and add tags
task = scenario.simulate(simulation_time_ns=1, tags=['experiment_1', 'molecule_1'])
task.add_tags(['experiment_2'])

# Retrieve the tags associated with the task
print(task.tags)  # or task.get_tags()
# Output: ['experiment_1', 'molecule_1', 'experiment_2']

# Fetch tasks based on tags
tasks = inductiva.tasks.get(last_n=10, status="success", tags=['experiment_1'])
```

### Implementation Explanation:

1. When creating a new task, users can specify tags to associate with it.
2. The `task` object has an attribute called `tags`, which allows users to add more tags using `task.add_tags()`.
3. Users can retrieve tasks based on specific tags, as demonstrated by the `inductiva.tasks.get` function.

### Future Enhancements:

In future iterations, we can enhance the tag search functionality, by including the ability to use logical statements for more complex tag searches. For example:

```python
tasks = inductiva.tasks.get(tags_selection="experiment_1 or experiment_2")
```
Do you agree with this structure?

**Comment by IvanPombo** [comment]
The examples you gave is what makes me wonder if we shouldn't have experiments/project first :)

Overall, I like the approach! Given that you are proposing this feature enhancement, does it mean that
```python
tasks = inductiva.tasks.get(last_n=10, status="success", tags=["experiment_1", "experiment_2"])
``` 

will fetch all tasks that possess both tags?

**Comment by cristianacarpinteiro** [comment]
> The examples you gave is what makes me wonder if we shouldn't have experiments/project first :)

I totally agree that we should implement experiments/projects, however, I think that this solution can fix it in the mean time. 

> Overall, I like the approach! Given that you are proposing this feature enhancement, does it mean that
> 
> ```python
> tasks = inductiva.tasks.get(last_n=10, status="success", tags=["experiment_1", "experiment_2"])
> ```
> 
> will fetch all tasks that possess both tags?

Yes and in a future iteration we implement the enhancement feature for more complex selections and we can do for example "experiment_1 and not experiment_2" and  "experiment_1 or experiment_2" .

**Comment by IvanPombo** [comment]
Thinking a bit more about this, I think the first stage is to change implement the `task.add_tags()`. This will just affect the tasks class. Otherwise, you will need to change all of the simulators and scenarios, which is more error prone. After this is implemented, we can play with it and check if it makes sense to add them on the simulate as well. In the latter case, it will much easier, because you can immediately do it with the `add_tags()` :)

**Comment by hpenedones** [comment]
I think that if you change the `simulate` method to accept tags, then this will only work for high-level scenarios, but not for low-level access to simulators. Why not just sticking to assigning tags to tasks with that line you have there:

```python
task.add_tags(['experiment_1, protein_1'])
```

**Comment by cristianacarpinteiro** [comment]
> I think that if you change the `simulate` method to accept tags, then this will only work for high-level scenarios, but not for low-level access to simulators. Why not just sticking to assigning tags to tasks with that line you have there:
> 
> ```python
> task.add_tags(['experiment_1, protein_1'])
> ```

Yes, I agree. I think that's the simplest and most efficient way of implementing it!
